examples/flutter_gallery/android/app/build.gradle

```diff
@@ -11,6 +11,14 @@ if (flutterRoot == null) {
     throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
 }
 
+def flutterVersion = file("$flutterRoot/version").text
+def flutterVersionComponents = flutterVersion.split(/[^0-9]+/)
+// Let the integer version code of xx.yy.zz-pre.nnn be xxyyzznnn.
+int flutterVersionCode = Math.min(flutterVersionComponents[0].toInteger(), 20) * 100000000
+flutterVersionCode += Math.min(flutterVersionComponents[1].toInteger(), 99) * 1000000
+flutterVersionCode += Math.min(flutterVersionComponents[2].toInteger(), 999) * 1000
+flutterVersionCode += Math.min(flutterVersionComponents[3]?.toInteger() ?: 0, 999)
```

Comment:
I'm seeing `flutter run` in flutter gallery failing on this line due to my version file having "unknown" instead of integer in flutterVersionComponents[3]:
```
╰─➤  cat $FH/flutter/version                                                                                                                                           0.0.0-unknown
╰─➤  $FH/flutter/bin/flutter doctor -v
[✓] Flutter (on Linux, locale en_US.UTF-8, channel master)
    • Flutter version 0.0.0-unknown at $FH/flutter
```

Reply:
Can you `flutter upgrade` and show us the full result of `flutter doctor` again? I wasn't expecting this version format on the master branch.